### PR TITLE
Log when capturing unfinished transaction

### LINKED ISSUE #1208: Check if transaction is finished during captureTransaction call
If the transaction is captured but not finished yet, what to do?
I'd say let's log it as INFO, finish and capture it.

Ideally `SentryClient.captureTransaction` does that as every call forwards to this method.

Context: https://github.com/getsentry/sentry-dotnet/pull/758#discussion_r564125517

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Log warning when capturing unfinished transaction.

## :bulb: Motivation and Context

Fixes #1208

## :green_heart: How did you test it?

Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [x] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -353,22 +353,29 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
...
-            buildEnvelope(sentryTransaction, filterForTransaction(getAttachmentsFromScope(scope)));
-        if (envelope != null) {
-          transport.send(envelope, hint);
-        } else {
+      final SentryTransaction sentryTransaction = (SentryTransaction) transaction;
+
+      if (!sentryTransaction.isFinished()) {
+        options.getLogger().log(SentryLevel.INFO, "Finishing unfinished transaction %s.", sentryId);
+        sentryTransaction.finish();
+      } else {
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryClient.java:361]: is that right? wouldn't it be better to `finish` AND capture the transaction? or am I missing something?

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/SentryClient.java:361]: ahhhh `finish` calls `captureTransaction` already, guess that's the trick, right?

**Comment by marandaneto** [comment]
@maciejwalkowiak changelog

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
I thought we had just agreed not to do this

**Comment by bruno-garcia** [comment]
The rationale against this change is that it'll be hiding a potential programming error. A transaction is expected to be finished through `finish()` (of if captured explicitly, have its close timestamp be set explicitly via an API) so we can avoid odd bugs where transactions look normal but have been captured before they were properly finished.

**Comment by marandaneto** [comment]
> I thought we had just agreed not to do this

> commented 4 hours ago

**Comment by maciejwalkowiak** [comment]
PR changed to include just logging a warning. I think we don't need changelog here.

File: sentry/src/main/java/io/sentry/ITransaction.java [hunk 2]
```diff
@@ -75,4 +75,11 @@
   @Nullable
   @ApiStatus.Internal
   String getTransaction();
+
+  /**
+   * Returns if transaction has finished.
+   *
+   * @return if transaction has finished.
+   */
+  boolean isFinished();
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/ITransaction.java:84]: should this go to `ISpan` instead?

Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/ITransaction.java]: Thats true, fixed.

**Review by marandaneto — APPROVED** [review]

**Comment by bruno-garcia** [comment]
> > I thought we had just agreed not to do this
> 
> commented 4 hours ago :) so even before the discussion on discord

Right, sorry about that

**Pull request merged**